scripts/ctest/ctest_driver.py

```diff
@@ -49,7 +87,9 @@ def run_single_test_suite(suite, ctest_path, cmake_build_path, build_config, dis
     :param only_gpu: optional, run only gpu-required tests
     :param generate_xml: optional, enable to produce the CTest xml file
     :param repeat: optional, number of times to run the tests in the suite
+    :param only_failed: optional, only usable with the repeat option. Run only initially failed tests during repeat
     :param extrargs: optional, forward args to ctest
+    :param soft_repeat_timeout: optional, only usable with the repeat option. Sets a soft timeout for the total test runtime
```

Comment:
What exactly is a soft timeout?

Reply:
I didn't want a hard timeout where it cuts off the test iteration mid run, so a soft timeout will essentially say "don't start a new iteration of repeat runs if we've exceeded our timeout, but finish the current iteration"

Reply:
I think it would be helpful to explain that somewhere. I would have never guessed that's what a soft timeout meant.